src/retry-strategy/unauthenticated-retry-strategy.ts

```diff
@@ -0,0 +1,49 @@
+import { Logger } from '../logger';
+import { NetworkRequest } from '../request';
+import { BaseSubscription } from '../subscription/base-subscription';
+import { RetryStrategy } from './retry-strategy';
+
+/**
+ * This serves as a no-op implementation of RetryStrategy that just relays whatever it has to the underlying requests. 
+ * Used with ExponentialBackoffRetryStrategy when we don't have a TokenProvider
+ */
+export class UnauthenticatedRetryStrategy implements RetryStrategy {
+    constructor(
+        private logger: Logger
+    ){}
+
+    private subscription: BaseSubscription;
```

Comment:
The retry strategy interface seems to imply the strategies are stateless. They are a hybrid in reality - the execute methods take callbacks, so I'd assume I could call `executeSubscription` multiple times, but it would lose the state of the all but last calls.